### PR TITLE
added missing setText method for Sign block

### PR DESCRIPTION
**Pull request by Frago9876543210**
## Introduction
Now there is no way to change sign text without player. Well, you can only use block entity. However, this is marked as deprecated

### CODE REVIEW

**Comment by dktapps** [comment]
SignText is mutable, this is redundant.

**Comment by Frago9876543210** [comment]
Since e61a08a56bb71c77c49cb070feeefa83e86397a2 SignText is immutable and this PR make sense

File: src/block/Sign.php [hunk 1]
```diff
@@ -142,6 +142,14 @@ public function getText() : SignText{
 		return $this->text;
 	}
 
+	/**
+	 * @param SignText $text
+	 */
+	public function setText(SignText $text) : void{
+		$this->text = $text;
+		$this->pos->getWorld()->setBlock($this->pos, $this);
```
Comment (dktapps) [inline src/block/Sign.php:147]: please remove this, it's possible that we may want to set the text of a block which doesn't have a position.

File: src/block/Sign.php [hunk 2]
```diff
@@ -165,8 +173,7 @@ public function updateText(Player $author, SignText $text) : bool{
 		}, $text->getLines())));
 		$ev->call();
 		if(!$ev->isCancelled()){
-			$this->text = clone $ev->getNewText();
-			$this->pos->getWorld()->setBlock($this->pos, $this);
```
Comment (dktapps) [inline src/block/Sign.php:169]: please restore the setBlock() call here

**Pull request merged**